lib/src/Uncertainty/Bayesian/MetropolisHastingsImplementation.cxx

```diff
@@ -192,31 +181,28 @@ Point MetropolisHastingsImplementation::getRealization() const
       throw InvalidArgumentException(HERE) << "The initial state should have non-zero posterior probability density";
   }
 
-  // for each new sample
-  for (UnsignedInteger i = 0; i < size; ++ i)
+  Point newState(currentState_);
+  const Point candidate(getCandidate());
+  for (UnsignedInteger j = 0; j < marginalIndices_.getSize(); ++ j)
+    newState[marginalIndices_[j]] = candidate[j];
+
+  const Scalar newLogPosterior = computeLogPosterior(newState);
+
+  // alpha = posterior(newstate)/posterior(oldstate)
```

Comment:
Looks like intermediate debug code to me.
```suggestion
```

Reply:
I think it was originally intended as an explanation of the next line rather than debug code, but it is false at any rate so I agree it should be removed.